proxy/flags.go

```diff
@@ -0,0 +1,210 @@
+		case "-H", "--host",
+			"-tlscacert", "--tlscacert",
+			"-tlscert", "--tlscert",
+			"-tlskey", "--tlskey":
+			filteredArgs = append(filteredArgs, args[i])
+			if !strings.Contains(args[i], "=") {
+				if i++; i < len(args) {
+					filteredArgs = append(filteredArgs, args[i])
+				}
+			}
+		}
+	}
+
+	var justVersion bool
+	var logLevel string
+	f := mflag.NewFlagSet("weaveproxy", mflag.ContinueOnError)
+	addFlags(f, c, &justVersion, &logLevel)
+	if err := f.Parse(filteredArgs); err != nil {
+		return err
+	}
+
+	newAddrs := []string{}
+	for _, addr := range c.ListenAddrs {
+		if strings.HasPrefix(addr, "unix://") {
+			addr = strings.Replace(addr, "docker", "weave", -1)
+		} else {
+			host, _, err := net.SplitHostPort(strings.TrimPrefix(addr, "tcp://"))
+			if err != nil {
+				return err
+			}
+			addr = fmt.Sprintf("tcp://%s:12375", host)
+		}
+		newAddrs = append(newAddrs, addr)
+	}
+	c.ListenAddrs = newAddrs
+
+	return nil
+}
+
+func findDocker(procfs, pid string) (dockerPID string, args []string, err error) {
+	statusFile, err := os.Open(filepath.Join(procfs, pid, "status"))
+	if err != nil {
+		return "", nil, err
+	}
+	defer statusFile.Close()
+	scanner := bufio.NewScanner(statusFile)
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		if len(fields) == 2 && fields[0] == "PPid:" {
+			dockerPID = fields[1]
+			break
+		}
+	}
+	if err = scanner.Err(); err != nil {
+		return
+	}
+
+	argBytes, err := ioutil.ReadFile(filepath.Join(procfs, dockerPID, "cmdline"))
+	if err != nil {
+		return "", nil, err
+	}
+
+	args = nullTermToStrings(argBytes)[1:]
+	if len(args) == 0 || args[0] == "/home/weave/weave" {
+		return findDocker(procfs, dockerPID)
+	}
+
+	return dockerPID, args, nil
+}
+
+func parseDockerEnv(procfs, dockerPID string, c *Config) error {
+	var configDir, dockerCertPath string
+
+	envBytes, err := ioutil.ReadFile(filepath.Join(procfs, dockerPID, "environ"))
+	if err != nil {
+		return err
+	}
+	for _, line := range nullTermToStrings(envBytes) {
+		switch {
+		case strings.HasPrefix(line, "DOCKER_CONFIG="):
+			if s := strings.SplitN(line, "=", 2); len(s) > 1 {
+				configDir = s[1]
+			}
+		case strings.HasPrefix(line, "DOCKER_CERT_PATH="):
+			if s := strings.SplitN(line, "=", 2); len(s) > 1 {
+				dockerCertPath = s[1]
+			}
+		case strings.HasPrefix(line, "DOCKER_TLS_VERIFY="):
+			c.TLSConfig.Verify = true
+		}
+	}
+
+	if configDir == "" {
+		configDir = filepath.Join(homedir.Get(), ".docker")
+	}
+	if dockerCertPath == "" {
+		dockerCertPath = configDir
+	}
+
+	c.TLSConfig.CACert = filepath.Join(dockerCertPath, "ca.pem")
+	c.TLSConfig.Cert = filepath.Join(dockerCertPath, "cert.pem")
+	c.TLSConfig.Key = filepath.Join(dockerCertPath, "key.pem")
+	return nil
+}
+
+func nullTermToStrings(b []byte) []string {
+	strs := []string{}
+	for {
+		i := bytes.IndexByte(b, 0)
+		if i == -1 {
+			break
+		}
+		strs = append(strs, string(b[0:i]))
+		b = b[i+1:]
+	}
+	return strs
+}
```

Comment:
(and looking at the call site, you could actually just use `strings.Split(string(b), "\000")`, I think)